### DISCUSSION TITLE
writing optimizers for use within gluon

### DISCUSSION THREAD

**Discussion by brightening-eyes** [apache/incubator-mxnet#20304] [Q&A] [answered]
dear everyone,
I want to write [this](https://github.com/mhaut/AngularGrad) which is written for pytorch. I am planning to code it in mxnet. is there any specific way where I can impliment this for mxnet?
thanks

**Comment by TristonC** [comment, marked as answer]
Thank you @brightening-eyes  for considering contributing. [This](https://github.com/apache/incubator-mxnet/pull/20065/files) might be a good example for adding an optimizer in MXNet.

**Reply by TristonC** [reply]
Please let us know if you running into any issues during the process. Looking forward to your PR.
